force-app/main/default/classes/DegreeTriggerHelper.cls

```diff
@@ -0,0 +1,124 @@
+public with sharing class DegreeTriggerHelper {
+  //  public static void preventUnrelatedClassSubject(List<ClassEnrollment__c> enrollmentBeingInserted) {
+  //      checkIfClassSubjectMatchesStudentMajor(enrollmentBeingInserted);
+  //  }
+
+    public static void preventUnrelatedClassSubject(ClassEnrollment__c enrollmentBeingInserted) {
+        checkIfClassSubjectMatchesStudentMajor(enrollmentBeingInserted);
+    }
+
+    static void checkIfClassSubjectMatchesStudentMajor(ClassEnrollment__c newEnrollment){
+        Id studentId = newEnrollment.Student__c;
+        Id classId = newEnrollment.Class__c;
+
+        Class__c classDept = [SELECT Department__r.Name FROM Class__c WHERE Id =: classId];
+        Student__c studentDept = [SELECT (SELECT Degree__r.Department__r.Name FROM Majors__r) FROM Student__c WHERE Id =: studentId];
+
+        if (studentDept.get('Name') != classDept.get('Name')){
+            newEnrollment.Status__c = 'Denied';
+            emailStudent();
+            System.debug('Wrong class subject');
+        }
+        else {
+            newEnrollment.Status__c = 'Enrolled';
+            emailTeacher();
+        }
+    }
+
+    // Taken from the Apex Developer Guide
```

Comment:
No need to rewrite the documentation. Paste a link to the article/page so that the user can view it if necessary.